// @crossmarkio/typings/build/src/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.latest = exports.Events = exports.Extension = exports.Models = exports.schema = exports.crossmark = exports.extension = void 0;
const extension = __importStar(require("./extension/index"));
exports.extension = extension;
const crossmark = __importStar(require("./crossmark/index"));
exports.crossmark = crossmark;
const schema = __importStar(require("./schema/index"));
exports.schema = schema;
const models_1 = require("./crossmark/models");
Object.defineProperty(exports, "Models", { enumerable: true, get: function () { return models_1.Models; } });
const index_1 = require("./extension/index");
Object.defineProperty(exports, "Extension", { enumerable: true, get: function () { return index_1.Extension; } });
const events_1 = require("./crossmark/events");
Object.defineProperty(exports, "Events", { enumerable: true, get: function () { return events_1.Events; } });
const index_2 = require("./schema/index");
Object.defineProperty(exports, "latest", { enumerable: true, get: function () { return index_2.latest; } });
exports.default = { extension, crossmark, schema };
